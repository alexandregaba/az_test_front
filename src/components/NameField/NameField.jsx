import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

const EditField = styled.input`
  margin: 3px 0 0;
  line-height: inherit;
  font-size: inherit;
  font-weight: ${props => (props.checked ? 'inherit' : 'bold')};
  color: ${props => (props.checked ? '#46CD7C' : 'inherit')};
  text-decoration: ${props => (props.checked ? 'line-through' : 'inherit')};
  width: 100%;
  height: 30px;
`;

const InnerSpan = styled.span`
  margin: 0 0 0 3px;
  align-self: flex-end;
  padding: 1px;
  color: ${props => (props.checked ? '#46CD7C' : 'inherit')};
  text-decoration: ${props => (props.checked ? 'line-through' : 'inherit')};
  font-weight: ${props => (props.checked ? 'inherit' : 'bold')};
  min-width: 100%;
  line-height: inherit;
  font-size: inherit;
  height: 30px;
`;

const NameField = ({ name, value, editMode, handleInputChange, onPressEnter, ...props }) => {
  if (editMode) {
    return (
      <EditField
        className="edit-field"
        ref={input => input && input.focus()}
        value={value}
        onChange={handleInputChange}
        onKeyPress={onPressEnter}
        {...props}
      />
    );
  }
  return (
    <InnerSpan className="item-name" {...props}>
      {name}
    </InnerSpan>
  );
};
NameField.propTypes = {
  name: PropTypes.string,
  value: PropTypes.string,
  checked: PropTypes.bool,
  editMode: PropTypes.bool,
  handleInputChange: PropTypes.func,
  onPressEnter: PropTypes.func,
};
export default NameField;
