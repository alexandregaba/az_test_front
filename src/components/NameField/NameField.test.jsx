import React from 'react';
import NameField from './NameField';
import { shallow } from 'enzyme';

const props = {
  name: 'orange',
  value: 'orange',
  editMode: false,
  checked: false,
  handleInputChange: jest.fn(),
  onPressEnter: jest.fn(),
};

it('matches snapshot', () => {
  const Component = shallow(<NameField {...props} />);
  expect(Component).toMatchSnapshot();
});
it('render span or input depending of editMode', () => {
  const Component = shallow(<NameField {...props} />);
  expect(Component.find('.item-name').length).toBe(1);

  const EditComponent = shallow(<NameField {...props} editMode />);
  expect(EditComponent.find('.edit-field').length).toBe(1);
});

it('use handleChange on changes', () => {
  const EditComponent = shallow(<NameField {...props} editMode />);
  EditComponent.find('.edit-field').simulate('change', { target: { value: 'Apple' } });
  expect(props.handleInputChange).toHaveBeenCalled();
});

it('use onPressEnter on enter', () => {
  const EditComponent = shallow(<NameField {...props} editMode />);
  EditComponent.find('.edit-field').simulate('keypress', { key: 'enter' });
  expect(props.onPressEnter).toHaveBeenCalled();
});
