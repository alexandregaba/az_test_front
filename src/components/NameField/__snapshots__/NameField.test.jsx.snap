// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<styled.span
  checked={false}
  className="item-name"
>
  orange
</styled.span>
`;
